loopchain/consensus/syncer.py

```diff
@@ -0,0 +1,144 @@
+            await self._raise_event()
+
+    async def _request_height(self):
+        peer_stub = self._stub_list[self._target]
+
+        response = peer_stub.BlockHeightRequest(loopchain_pb2.HeightRequest(
+            peer=ChannelProperty().peer_target,
+            channel=self._block_manager.channel_name
+        ), conf.GRPC_TIMEOUT)
+
+    async def _request_block(self):
+        def _request(idx: int, height: int):
+            peer_stub = self._stub_list[idx]
+
+            response = peer_stub.BlockRequest(loopchain_pb2.PeerHeight(
+                peer=ChannelProperty().peer_target,
+                channel=self._block_manager.channel_name,
+                height=height
+            ), conf.GRPC_TIMEOUT)
+
+        max_loop = self._max_height_in_nodes-3
+        if self._last_block_height < max_loop:
+            for i in range(1, min(conf.CITIZEN_ASYNC_RESULT_MAX_SIZE+1, max_loop-self._last_block_height)):
+                height = self._last_block_height + i
+                if height in self._request_history_list.keys():
+                    during_request_time = time.time()-self._request_history_list[height][1]
+                    if during_request_time > conf.WAIT_SUB_PROCESS_RETRY_TIMES:
+                        retry_target_idx = self._request_history_list[height][0]
+                        _request(retry_target_idx, height)
+                        self._request_history_list[height] = [retry_target_idx, time.time()]
+                else:
+                    _request(self._target_idx, height)
+                    self._request_history_list[height] = list()
+                    self._request_history_list[height] = [self._target_idx, time.time()]
+                    self._target_idx = (self._target_idx+1) % len(self._stub_list)
+
+                await asyncio.sleep(0)
+
+    def receive_vote(self, vote: 'BlockVote'):
+        height = vote.block_height
+        self._max_height_in_nodes = max(height, self._max_height_in_nodes)
+
+        if abs(self._last_block_height-height) < conf.CITIZEN_ASYNC_RESULT_MAX_SIZE:
+            if not height in self._vote_info_other_nodes.keys():
+                self._vote_info_other_nodes[height] = list()
+
+            self._vote_info_other_nodes[height].append(vote)
+
+
+    def receive_data(self, block_data: 'Block'):
+        height = block_data.header.height
+        self._max_height_in_nodes = max(height, self._max_height_in_nodes)
+
+        diff_height_info = abs(self._last_block_height-self._max_height_in_nodes)
+        if diff_height_info < conf.CITIZEN_ASYNC_RESULT_MAX_SIZE:
+            if not height in self._data_info_other_nodes.keys():
+                self._data_info_other_nodes[height] = list()
+
+            self._data_info_other_nodes[height].append(block_data)
+
+            if 3 < diff_height_info:
+                for vote in block_data.prev_votes:
+                    self.receive_vote(vote)
+
+    async def _raise_event(self):
+        for i in range(1, 3):
```

Comment:
How about adding some comments or wrapping by variable onto this line, to show your intension?